File: docspring-generator/src/main/resources/php/examples/generate_pdf.erb.php
<?php
$template_id = 'tpl_000000000000000001'; // string |
$submission_data = new \DocSpring\Model\SubmissionData();
$submission_data->setData([
  "title" => 'Test PDF',
  "description" => 'This is a test PDF!',
]);
$submission_data->setTest(<%= @test_pdfs %>);
$submission_data->setFieldOverrides([
  "title" => [
    "required" => false
  ]
]);
<% if @options.fetch(:type) == :original -%>
$response = $docspring->generatePDF($template_id, $submission_data);
$submission = $response->getSubmission();
<% else -%>
$submission = $docspring->generatePDF($template_id, $submission_data);
<% end -%>

<% if @template_type == :doc -%>
echo "Download your PDF at: " . $submission->getDownloadUrl();
<% end -%>

File: docspring-generator/src/main/resources/ruby-client/examples/generate_pdf.erb.rb
client = DocSpring::Client.new
template_id = 'tpl_000000000000000001'

<% if @template_type != :doc -%>
expect(client).to receive(:sleep).with(1).once
<% end -%>
response = client.generate_pdf(
  template_id: template_id,
  data: {
    title: 'Test PDF',
    description: 'This PDF is great!',
  },
  field_overrides: {
    title: {
      required: false
    }
  }
)
submission = response.submission

<% if @template_type == :doc -%>
puts "Download your PDF at: #{submission.download_url}"
<% else -%>
expect(response.status).to eq 'success'
expect(submission.id).to start_with 'sub_'
expect(submission.expired).to eq false
expect(submission.state).to eq 'processed'
expect(submission.pdf_hash).to eq 'TEST_SUBMISSION_SHA256_HASH'
expect(submission.download_url).to include 'submissions/submission.pdf'
<% end -%>
